esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,527 @@
+        process = subprocess.Popen(ncl_call,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT,
+                                   env=env)
+        output, err = process.communicate()
+        for oline in str(output.decode('utf-8')).split('\n'):
+            logger.info('[NCL] %s', oline)
+        if err:
+            logger.error('[NCL][subprocess.Popen ERROR] %s', err)
+            return False
+        return True
+
+    def _run_pyt_script(self, in_dir, out_dir, dataset, start, end):
+        """Run the Python cmorization mechanism."""
+        module_name = ('esmvaltool.cmorizers.data.formatters.datasets.' +
+                       dataset.lower().replace("-", "_"))
+        module = importlib.import_module(module_name)
+        logger.info("CMORizing dataset %s using Python script %s", dataset,
+                    module.__file__)
+        cmor_cfg = read_cmor_config(dataset)
+        module.cmorization(in_dir, out_dir, cmor_cfg, self.config, start, end)
+        logger.info('CMORization of dataset %s finished!')
+        return True
+
+
+class DataCommand():
+    """Download and format data to use with ESMValTool."""
+    def __init__(self):
+        datasets_file = os.path.join(os.path.dirname(__file__), 'datasets.yml')
+        with open(datasets_file) as data:
+            self._info = yaml.safe_load(data)
+        self.formatter = Formatter(self._info)
+
+    def list(self):
+        """List all supported datasets."""
+        print()
+        print(f'| {"Dataset name":30} | Tier | Auto-download | Last access |')
+        print('-' * 71)
+        for dataset, dataset_info in self._info['datasets'].items():
+            date = datetime.datetime.strptime(str(dataset_info['last_access']),
+                                              "%Y-%m-%d")
+            print(f"| {dataset:30} | {dataset_info['tier']:4} "
+                  f"| {self.formatter.has_downloader(dataset):13} "
+                  f"|  {date.strftime('%Y-%m-%d')} |")
+        print('-' * 71)
+
+    def info(self, dataset):
+        """Show detailed info about a specific dataset.
+
+        Parameters
+        ----------
+        dataset : str
+            dataset to show
+        """
+        dataset_info = self._info['datasets'][dataset]
+        print(dataset)
+        print()
+        print(f"Tier: {dataset_info['tier']}")
+        print(f"Source: {dataset_info['source']}")
+        print(f"Automatic download: {self.formatter.has_downloader(dataset)}")
+        print("")
+        print(dataset_info['info'])
+
+    def download(self,
+                 datasets,
+                 config_file=None,
+                 start=None,
+                 end=None,
+                 overwrite=False,
+                 **kwargs):
+        """Download datasets.
+
+        Parameters
+        ----------
+        datasets : list(str), optional
+            List of datasets to format, by default None
```

Comment:
Currently, `datasets` is not optional, which is better imo. So the description should be adjusted.